molgenis-data-rest/src/main/java/org/molgenis/data/rest/RestController.java

```diff
@@ -735,7 +739,14 @@ public void delete(@PathVariable("entityName") String entityName, @PathVariable(
 		EntityType entityType = dataService.getEntityType(entityName);
 		Object id = getTypedValue(untypedId, entityType.getIdAttribute());
 
-		dataService.deleteById(entityName, id);
+		if (ATTRIBUTE_META_DATA.equals(entityName))
```

Comment:
Metadata == data, the REST controller shouldn't make any entity specific exceptions.